gpt_index/readers/chroma.py

```diff
@@ -31,41 +34,103 @@ def __init__(
+            raise ValueError("Please provide a collection name.")
         from chromadb.config import Settings
 
-        self._client = chromadb.Client(
-            Settings(
-                chroma_db_impl="duckdb+parquet", persist_directory=persist_directory
+        if persist_directory:
+            self._client = chromadb.Client(
+                Settings(
+                    chroma_db_impl="duckdb+parquet", persist_directory=persist_directory
+                )
+            )
+        else:
+            self._client = chromadb.Client(
+                Settings(
+                    chroma_api_impl="rest",
+                    chroma_server_host=host,
+                    chroma_server_http_port=port,
+                )
             )
-        )
         self._collection = self._client.get_collection(collection_name)
 
-    def load_data(
-        self,
-        query_vector: Any,
-        limit: int = 10,
-    ) -> Any:
-        """Load data from Chroma.
+    def create_documents(self, results: tuple) -> List[Document]:
+        """Create documents from the results.
 
         Args:
-            query_vector (Any): Query
-            limit (int): Number of results to return.
+            results: Results from the query.
 
         Returns:
-            List[Document]: A list of documents.
+            List of documents.
         """
-        results = self._collection.query(query_embeddings=query_vector, n_results=limit)
-
         documents = []
-        for result in zip(results["ids"], results["documents"], results["embeddings"]):
+        for result in zip(
+            results["ids"],
+            results["documents"],
+            results["embeddings"],
+            results["metadatas"],
+        ):
             document = Document(
                 doc_id=result[0][0],
                 text=result[1][0],
                 embedding=result[2][0],
+                extra_info=result[3][0],
             )
             documents.append(document)
 
         return documents
+
+    def load_embedding_data(
+        self,
+        query_embedding: List[float],
+        limit: int = 10,
+        where: dict = {},
+        where_document: dict = {},
+    ) -> Any:
+        """Load data from the collection.
+
+        Args:
+            query_embedding: Query embedding.
+            limit: Number of results to return.
+            where: Filter results by metadata. {"metadata_field": "is_equal_to_this"}
+            where_document: Filter results by document. {"$contains":"search_string"}
+
+        Returns:
+            List of documents.
+        """
+        results = self._collection.query(
+            query_embeddings=[query_embedding],
+            n_results=limit,
+            where=where,
+            where_document=where_document,
+            include=["metadatas", "documents", "distances", "embeddings"],
+        )
+        return self.create_documents(results)
+
+    def load_data(
+        self,
```

Comment:
this will break backwards compatibility with the existing query function. can you include `query_vector` as the second position (and still keep query_texts as a separate arg? kwarg maybe?). Then load_embedding_data may not be necessary either

Reply:
Thanks, made updates as requested to remove any backwards compatible breaking changes.